File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        j = 0
        k =1
        maxP = 0
        while k < len(prices):
            p = prices[k]-prices[j]
            maxP=max(maxP,p)
            if prices[k] < prices[j]:
                j = k
            k=k+1
        return maxP
        

File: Data Structures & Algorithms/max-water-container/submission-2.java
class Solution {
    public int maxArea(int[] heights) {
        int max_A = 0;
        int l = 0;
        int r = heights.length - 1;
        while (l < r){
            int wid = r - l;
            int h = Math.min(heights[l],heights[r]);
            int area = wid * h;
            max_A = Math.max(max_A,area);
            if (heights[l] < heights[r]){
               l = l + 1;
            }
            else{
                r = r - 1;
            }
        }
        return max_A;
    }
}
